mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,376 @@
+
+.. literalinclude::  authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/Home/ContactCopy.cshtml
+   :language: aspx-cs
+   :emphasize-lines: 15-16
+   :lines: 1-17
+ 
+4. Run the app and verify that it generates the correct links.
+
+**Note:** If you were to write the email tag self-closing (``<email mail-to="Rick" />``), the final output would also be self-closing. In our example, the output would be ``<a href="mailto:Rick@contoso.com" />``. Self-closing anchor tags (also known as `void elements <http://www.w3.org/TR/html5/syntax.html#void-elements>`__) are not valid HTML, so you wouldn't want to create one, but you might want to create a tag helper for a void element. The ASP.NET 5 runtime sets the type of the ``TagMode`` property after reading a tag.
+
+
+In this section we will update the ``EmailTagHelper`` so that it gets the target ``mail-to`` from the content. 
+
+#. Update the ``EmailTagHelper`` class with the following code:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/EmailTagHelper.cs
+   :lines: 6-18
+   :dedent: 3
+
+Notes:
+
+- We use the ``context`` parameter to get contents of the HTML element.
+- The following line shows the syntax for adding attributes:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/EmailTagHelper.cs
+   :lines: 15
+   :dedent: 12
+   
+That approach works for the attribute "href" as long as it doesn't currently exit in the attributes collection. You can also use the ``output.Attributes.Add`` method to add a tag helper attribute to the end of the collection of tag attributes.
+
+
+- The last line sets the completed content for our minimally functional tag helper.
+- This version uses the asynchronous ``ProcessAsync`` method. The asynchronous `GetChildContentAsync <https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContext.cs#L65-L79>`__ returns a ``Task`` containing the `TagHelperContent <https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContent.cs>`__. 
+
+2. Run the app and verify that it generates  a valid email link.
+    
+
+The bold tag helper
+---------------------------
+
+#. Add the following ``BoldTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/BoldTagHelper.cs
+    :language: c#
+
+Notes:
+
+- The ``[TargetElement]`` attribute passes an attribute parameter that specifies that any HTML element that contains an HTML attribute value of "bold" will match, and the ``Process`` override method in the class will run. In our sample,  the ``Process``  method removes the "bold" attribute value and surrounds the containing markup with ``<strong></strong>``.
+- Because we are keeping the tag content, we must write the opening tag with the ``PreContent.SetContent`` method and the closing tag with the ``PostContent.SetContent`` method.
```

Comment:
`Because we are keeping the tag content` => `Because we don't want to replace the existing tag content`